various-receipes/error-handling/test/error-handling.tests.js

```diff
@@ -0,0 +1,139 @@
+const request = require("supertest");
+const sinon = require("sinon");
+const nock = require("nock");
+const { initializeWebServer, stopWebServer } = require("../../../example-application/api-under-test");
+const OrderRepository = require("../../../example-application/data-access/order-repository");
+const { metricsExporter } = require("../../../example-application/error-handling");
+const { AppError } = require("../../../example-application/error-handling");
+
+let expressApp;
+
+beforeAll(async (done) => {
+  // ️️️✅ Best Practice: Place the backend under test within the same process
+  expressApp = await initializeWebServer();
+  // ️️️✅ Best Practice: Ensure that this component is isolated by preventing unknown calls except for the Api-Under-Test
+  nock.disableNetConnect();
+  nock.enableNetConnect("127.0.0.1");
+
+  done();
+});
+
+afterAll(async (done) => {
+  // ️️️✅ Best Practice: Clean-up resources after each run
+  await stopWebServer();
+  done();
+});
+
+beforeEach(() => {
+  nock.cleanAll();
+  // ️️️✅ Best Practice: Isolate the service under test by intercepting requests to 3rd party services
+  nock("http://localhost/user/").get(`/1`).reply(200, {
+    id: 1,
+    name: "John",
+  });
+  nock("https://mailer.com").post("/send").reply(202);
+  sinon.restore();
```

Comment:
Move to `afterEach`? (also the `nock.cleanAll`)

Reply:
Done